src/classes/RD2_OpportunityEvaluationService.cls

```diff
@@ -168,10 +179,19 @@ public inherited sharing class RD2_OpportunityEvaluationService {
      */
     @TestVisible
     private Boolean isNewRecurringDonation(npe03__Recurring_Donation__c rd) {
+        Boolean noOpenOpp = true;
+
+        for (Opportunity opp : rd.npe03__Donations__r) {
+            if (opp.IsClosed == false || opp.CloseDate >= Date.today()) {
+                noOpenOpp = false;
+                break;
+            }
+        }
+
         return
             rd.npe03__Next_Payment_Date__c != null
             && rd.Status__c == RD2_Constants.STATUS_ACTIVE 
-            && rd.npe03__Donations__r.isEmpty()
+            && noOpenOpp
```

Comment:
New RD is record not having any Opp regardless of the Close Date or Stage. Thus, I think the previous check for empty list was fine. What is the reason to check for Opp details for new RDs?

Reply:
Before the query on opp is only querying opp that is not closed or the closeDate is in the future. I modify the query to query all opps under the rd in order to determine if it's the first installment for the rd. This extra detail check is moving the query part of the filter into the method